projects/RabbitMQ.Client/RabbitMQ.Client.csproj

```diff
@@ -61,8 +61,12 @@
     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.2" PrivateAssets="All" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="All" />
     <PackageReference Include="MinVer" Version="3.1.0" PrivateAssets="All" />
+    <PackageReference Include="OpenTelemetry.Api" Version="1.3.1" />
```

Comment:
Having this in the default package, means everyone has now a dependency to it. Is this something we want to have?

Would it be an alternative to have this instead of in the base as an additional package that you can add? (Will have to have a different implementation then obviously)

Reply:
Good point. @cijothomas is there a best practice here?

If we do not have this build directly into the client, we'd need to create extensibility points where this could be injected, taking into account how we maintain Activity and async contexts to maintain as much sanity as possible in the traces.

Reply:
Seems this is in accordance with the library guidelines here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/library-guidelines.md

Reply:
This is the normal practice for all opentelemetry implementations. In case of .NET, since most of the API is part of the System.Diagnostics.DiagnosticSource package itself, one may just take a dependency on that and avoid OpenTelemetry.Api dependency. In this library, you'll need the .API package to get the propagation part. (.NET do have propagators added, it might be an alternative solution.)

Reply:
Note that there is currently an effort to explore moving this over to the new-ish `DistributedContextPropagator` to replace the need for libraries to reference OpenTelemetry.Api:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/3765

Reply:
Thanks for the tip @rwkarg. Rebased and updated!